docs/spec/baseapp/WIP_abci_application.md

```diff
@@ -0,0 +1,46 @@
+# ABCI application
```

Comment:
I wonder if we should combine this with https://github.com/cosmos/cosmos-sdk/pull/2781/files#diff-869e0cb8376caf51500c202067bcd9aa

Reply:
We definitely should...somehow.

Reply:
Looks like this doc lives in a new location, so we'll have to update that I suppose.

Reply:
meh, these are two different (both incomplete) docs for different purposes, the spec is unique from the reference material, we need them both. I just chose to include updates in the spec. I don't think this level of granularity in the docs need-be destined for reference material - it seems like a spec level of detail.

Reply:
Mhmm I have to disagree here. The doc is incomplete yes, but with there is already major overlap. In fact that doc has info on checkTx, deliverTx, and even maximum signature logic.

Reply:
yeah maybe some of the content from the reference document should also be moved into the spec then. But we do need distinct documents here - they serve different purposes (but the reference document should also refer users to the spec for greater level of detail). Not going to deal as a part of this PR - this is part of a much larger docs refactor which is required